lib/chef/provisioning/aws_driver/aws_resource.rb

```diff
@@ -60,6 +60,23 @@ def aws_object
     raise NotImplementedError, :aws_object
   end
 
+
+  #
+  # Is this object ready to be tagged?
+  #
+  def taggable?
+    false
+  end
+
+  #
+  # Wait until this object.taggable? returns true
+  #
+  def wait_until_taggable()
+    Retryable.retryable(:tries => 12, :sleep => 5, :on => [AWS::EC2::Errors::InvalidInstanceID::NotFound, TimeoutError]) do
+      raise TimeoutError unless taggable?
```

Comment:
anything that subclasses `AWSResource` can override `taggable?`, and that method would be used in `wait_until_taggable()`